File: common/src/main/java/generations/gg/generations/core/generationscore/common/network/packets/statue/S2COpenStatueEditorScreenPacket.java
package generations.gg.generations.core.generationscore.common.network.packets.statue;

import generations.gg.generations.core.generationscore.common.GenerationsCore;
import generations.gg.generations.core.generationscore.common.GenerationsCore;
import generations.gg.generations.core.generationscore.common.GenerationsCore;
import generations.gg.generations.core.generationscore.common.network.packets.GenerationsNetworkPacket;
import generations.gg.generations.core.generationscore.common.network.packets.GenerationsNetworkPacket;
import generations.gg.generations.core.generationscore.common.network.packets.GenerationsNetworkPacket;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.resources.ResourceLocation;

public record S2COpenStatueEditorScreenPacket(int entityId) implements GenerationsNetworkPacket<S2COpenStatueEditorScreenPacket> {
    public static final ResourceLocation ID = GenerationsCore.id("open_statue_editor_screen");

    public static S2COpenStatueEditorScreenPacket decode(FriendlyByteBuf buf) {
        return new S2COpenStatueEditorScreenPacket(buf.readInt());
    }

    @Override
    public void encode(FriendlyByteBuf buf) {
        buf.writeInt(entityId());
    }

    @Override
    public ResourceLocation getId() {
        return ID;
    }

}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/network/packets/statue/S2COpenStatueEditorScreenPacket.kt
package generations.gg.generations.core.generationscore.common.network.packets.statue

import com.cobblemon.mod.common.api.net.NetworkPacket
import generations.gg.generations.core.generationscore.common.GenerationsCore
import net.minecraft.network.FriendlyByteBuf
import net.minecraft.network.RegistryFriendlyByteBuf
import net.minecraft.resources.ResourceLocation

class S2COpenStatueEditorScreenPacket(val entityId: Int) : NetworkPacket<S2COpenStatueEditorScreenPacket> {
    override val id: ResourceLocation = ID
    override fun encode(buf: RegistryFriendlyByteBuf) {
        buf.writeInt(entityId)
    }

    companion object {
        val ID: ResourceLocation = GenerationsCore.id("open_statue_editor_screen")

        fun decode(buf: RegistryFriendlyByteBuf): S2COpenStatueEditorScreenPacket {
            return S2COpenStatueEditorScreenPacket(buf.readInt())
        }
    }
}